### PR TITLE
Fix for ticket 12685

### PR DESCRIPTION
**Pull request by mthornhill**

### CODE REVIEW

File: tests/serializers/tests.py [hunk 1]
```diff
@@ -278,6 +278,12 @@ def test_deterministic_mapping_ordering(self):
             'second_category_pk': categories[1],
         })
 
+    def test_deserialize_force_insert(self):
+        """Tests that deserialized content can be saved with force_insert as a parameter."""
+        serial_str = serializers.serialize(self.serializer_name, Article.objects.all())
+        models = serializers.deserialize(self.serializer_name, serial_str)
+        [model.save(force_insert=False) for model in models]
```
Comment (charettes) [inline tests/serializers/tests.py]: I think an explicit `for` loop wouldn't hurt here.

Comment (charettes) [inline tests/serializers/tests.py]: Shouldn't this test fail if you're `force_insert`ing an object that already exists?

File: tests/serializers/tests.py [hunk 2]
```diff
@@ -278,6 +278,13 @@ def test_deterministic_mapping_ordering(self):
             'second_category_pk': categories[1],
         })
 
+    def test_deserialize_force_insert(self):
+        """Tests that deserialized content can be saved with force_insert as a parameter."""
+        serial_str = serializers.serialize(self.serializer_name, Article.objects.all())
+        models = serializers.deserialize(self.serializer_name, serial_str)
+        for model in models:
+            model.save(force_insert=False)
```
Comment (charettes) [inline tests/serializers/tests.py]: Never mind my previous comment, I read `force_insert=True` somehow.

File: tests/serializers/tests.py (hunk 1 above)
Comment (mthornhill) [inline tests/serializers/tests.py]: Hi Simon,
I've added an explicit for loop on branch
https://github.com/django/django/pull/4811/files and it does fail if
force_insert=True
I'm testing that we can pass the parameter more than the functionality of
force_insert.
Thanks
Michael

On Fri, Jun 5, 2015 at 4:11 PM, Simon Charette <notifications@github.com>
wrote:

> In tests/serializers/tests.py
> <https://github.com/django/django/pull/4811#discussion_r31820827>:
>
> > @@ -278,6 +278,12 @@ def test_deterministic_mapping_ordering(self):
> >              'second_category_pk': categories[1],
> >          })
> >
> > +    def test_deserialize_force_insert(self):
> > +        """Tests that deserialized content can be saved with force_insert as a parameter."""
> > +        serial_str = serializers.serialize(self.serializer_name, Article.objects.all())
> > +        models = serializers.deserialize(self.serializer_name, serial_str)
> > +        [model.save(force_insert=False) for model in models]
>
> I think an explicit for loop wouldn't hurt here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/4811/files#r31820827>.
>



-- 
p: +353 87 6124111
e: michael@maithu.com

**Comment by charettes** [comment]
Buildbot, test this please.

File: tests/serializers/tests.py (hunk 2 above)
Comment (freakboy3742) [inline tests/serializers/tests.py]: This test requires a little bit more validation - all this checks is that force_insert is a valid argument (or, at least, and argument to save() that doesn't raise an error). It doesn't verify that force_insert is passed into the underlying save method.

**Comment by mthornhill** [comment]
Hi Russell,
I've changed the test around a bit to mock django.db.models.Model so I can validate that save_base is called with the correct parameters, does this look like the correct way to go?
Thanks
Mick

**Comment by freakboy3742** [comment]
Hi Mick,

That should do the trick. If you can collapse this patch into a single commit, I'll merge it.

**Comment by mthornhill** [comment]
Hi Russell,
That's collapsed into one commit now, thanks for  you help.
Mick

**Pull request merged**